lib/puppet/network/http/webrick/rest.rb

```diff
@@ -15,8 +15,12 @@ def self.mutex
 
   def initialize(server)
     raise ArgumentError, "server is required" unless server
-    register([Puppet::Network::HTTP::API::V3.routes,
-              Puppet::Network::HTTP::API::V2.routes])
+    prefix = Regexp.new("^#{Puppet[:master_url_prefix]}")
```

Comment:
@cprice404 yup, that only affects the subsequent PR, and I've made that change there.